transmogrifier/sources/transformer.py

```diff
@@ -364,6 +369,30 @@ def get_optional_fields(self, _source_record: dict[str, JSON] | Tag) -> dict | N
         """
         return {}
 
+    @final
+    @staticmethod
+    def create_dates_and_locations_from_publishers(fields: dict) -> dict:
```

Comment:
This is specific to this use case but as we do more of these duplicate objects for facets, this can be refactored to be more generic

Reply:
I think this is an interesting case.

I'm on board with waiting to either make more generic, establish some kind of `post_transform_hooks` linkage, and/or refactor to use metadata field methods.

An example of how these things might be intertwined.  We could create decorators that are _optionally_ applied to field methods, that would do that work

```python
# post field method run decorators
def extract_dates():
    pass

def extract_locations():
    pass

def extract_subjects():
    pass

class MyTransformer():
    
    #specific field method getting those decorators
    @extract_dates
    @extract_locations
    def get_publishers(self):
        pass
```

Reply:
I think the decorator approach has promise whenever we tackle this